src/antelope/stores/balances.ts

```diff
@@ -142,17 +142,55 @@ export const useBalancesStore = defineStore(store_name, {
                 console.error('Error: ', error);
             }
         },
+        async updateSystemTokensPrices(label: string): Promise<void> {
+            this.trace('updateSystemTokensPrices', label);
+            try {
+                // take the three system tokens
+                const chain_settings = useChainStore().getChain(label).settings as EVMChainSettings;
+                const sysToken = chain_settings.getSystemToken();
+                const wrpToken = chain_settings.getWrappedSystemToken();
+                const stkToken = chain_settings.getStakedSystemToken();
+
+                // get the price for both system and wrapped tokens
+                const price = (await chain_settings.getUsdPrice()).toString();
+                const marketInfo = { price } as MarketSourceInfo;
+                sysToken.market = new TokenMarketData(marketInfo);
+                wrpToken.market = new TokenMarketData(marketInfo);
+
+                // first we need the contract instance to be able to execute queries
+                const evm = useEVMStore();
+                const authenticator = useAccountStore().getEVMAuthenticator(label);
+                const contract = await evm.getContract(authenticator, stkToken.address, stkToken.type);
+                if (!contract) {
+                    throw new AntelopeError('antelope.balances.error_token_contract_not_found', { address: stkToken.address });
+                }
+                const contractInstance = await contract.getContractInstance();
+
+                // Now we preview a deposit of 1 SYS to get the ratio
+                const oneSys = ethers.utils.parseUnits('1.0', sysToken.decimals);
+                const ratio:BigNumber = await contractInstance.previewDeposit(oneSys);
 
+                // Now we calculate the price of 1 STK = (price of 1 SYS) / ratio
+                const stkPrice = sysToken.price.value.mul(oneSys).div(ratio);
+                const stkPriceNumber = ethers.utils.formatUnits(stkPrice, sysToken.decimals);
```

Comment:
Using the `BigNumber` `mul` and `div` methods will lead to the loss of precision because `BigNumber` can't handle fractional (non-integer) values. I recommend using the `convertCurrency` util to get a more accurate number.
it would look something like:
```ts
const stakedTokenDecimals = await contractInstance.decimals();
const stkPrice = convertCurrency(oneSys, sysToken.decimals, stakedTokenDecimals, ratio);
```

Reply:
of course!
I forgot about that utility function. Thanks